assets/src/edit-story/components/panels/textStyle/padding.js

```diff
@@ -71,23 +72,16 @@ function PaddingControls({
 
   const handleChange = useCallback(
     (newPadding, submit = false) => {
-      pushUpdate(({ x, y, width, height }) => {
-        const updates = {};
-
-        if ('horizontal' in newPadding) {
-          updates.x = x - (newPadding.horizontal - padding.horizontal || 0);
-          updates.width =
-            width + (newPadding.horizontal - padding.horizontal || 0) * 2;
-        }
-
-        if ('vertical' in newPadding) {
-          updates.y = y - (newPadding.vertical - padding.vertical || 0);
-          updates.height =
-            height + (newPadding.vertical - padding.vertical || 0) * 2;
-        }
-
-        return updates;
-      });
+      pushUpdate(({ x, y, width, height }) =>
```

Comment:
Can you confirm the added padding is accurately reflected in the output as well?

Unfortunately it's hard to test right now due to a regression (#5324), however, can be done when adding the "mask" back for the text output temporarily, e.g. just undoing [this commit](https://github.com/google/web-stories-wp/pull/5123/commits/3a99d9f9ebf15ca40f3f00e8f80c02d31374d69b) for the time of testing.